staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -203,9 +205,11 @@ type udsGRPCConnector struct {
 // TODO At the moment, it does not use the provided context.
 func (u *udsGRPCConnector) connect(_ context.Context) (proxier, error) {
 	udsName := u.udsName
-	dialOption := grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+	dialOption := grpc.WithContextDialer(func(grpcDialCtx context.Context, addr string) (net.Conn, error) {
+		udsDialCtx, cancel := context.WithTimeout(grpcDialCtx, time.Duration(5*time.Second))
+		defer cancel()
 		var d net.Dialer
```

Comment:
same comment as in https://github.com/kubernetes/kubernetes/pull/113486/files#r1010969702